### PR TITLE
Profile followups 

### LINKED ISSUE #6972: Profile followups
## Top part
- i would like to change the create onchain identity button a bit
image
- its This button 
image
- 12 horizontal padding and 4px top-bottom 
- button is the same as in profile sidebar
- hover is black stroke 
image

## Activity tab 
- some changes here as well
- mainly reuses the collection activity part 
- we will rename gift to transfer because it is more general
image
- so it goes like this: item, event, amount, from, to, time
- Please remove the hover option over the row
- blue links have hover card - address from to
- item is clickable (preview and name cursor pointer) 
- for more visit any collection activity tab

### PR DESCRIPTION
**Pull request by hassnian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #6972
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-08-30 at 15 03 45@2x

CleanShot 2023-08-30 at 15 02 52@2x

### <samp>🤖 Generated by Copilot at b4265ec</samp>

This pull request introduces several changes to improve the UI and functionality of the activity history table in the profile and collection pages. It adds new components such as `EventTag`, `ResponsiveTable`, and `HistoryRow` to display the event data in a consistent and responsive way. It also refactors the existing components such as `History` and `EventRowDesktop` to use the new components and props. It updates the GraphQL query to fetch the metadata of the NFTs involved in the events and displays them in the table. It removes some unused code from the `utils/historyEvent` file.

### <samp>🤖 Generated by Copilot at b4265ec</samp>

> _We create and destroy with our code_
> _We unleash the power of the EventTag_
> _We make the table responsive and bold_
> _We query the NFTs and their metadata_

### CODE REVIEW

**Comment by hassnian** [comment]
@exezbcz no Percentage column ?

File: components/profile/ProfileDetail.vue [hunk 1]
```diff
@@ -36,9 +36,16 @@
               emit
               @change="handleIdentity" />
           </h1>
-          <nuxt-link v-if="isAllowSetIdentity" to="/identity">
+          <NeoButton
+            v-if="isAllowSetIdentity"
+            href="/identity"
+            no-shadow
+            rounded
+            tag="a"
```
Comment (roiLeo) [inline components/profile/ProfileDetail.vue:44]:
```suggestion
            v-if="isAllowSetIdentity"
            to="/identity"
            no-shadow
            rounded
            tag="nuxt-link"
```

File: components/profile/activityTab/HistoryRow.vue [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+.fixed-height {
+  height: 22px;
+}
+.height-50px {
+  height: 50px;
+}
+
+.height-70px {
+  height: 70px;
+}
```
Comment (roiLeo) [inline components/profile/activityTab/HistoryRow.vue:213]: can't do otherwise? maybe change it to `h-70`? use `rem`?

**Comment by exezbcz** [comment]
> @exezbcz pls do visual check
> 
> also no Percentage column ?

I dont think its necessary, most of the times you will see 100% there (sold at list price after mint), we have holders and flipper tab for that

**Comment by exezbcz** [comment]
Please add 16px bottom margin on the create identity button  
image

**Comment by hassnian** [comment]
@exezbcz pls check now

File: components/profile/activityTab/HistoryRow.vue [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  return props.event.Type
+})
+
+const interactionName = computed(
+  () => interactionNameMap[eventType.value] || eventType.value
+)
+
+onMounted(async () => {
+  parseNftAvatar(props.event.Item).then((response) => (avatar.value = response))
+})
```
Comment (roiLeo) [inline components/profile/activityTab/HistoryRow.vue:173]: I would use 
```ts
const getAvatar = async () => {
  if (props.nft) {
    avatar.value = await parseNftAvatar(props.nft)
  }
}
```
and call `getAvatar` onMounted

**Review by roiLeo — APPROVED** [review]
otherwise lgtm

**Comment by exezbcz** [comment]
@hassnian works for me, thanks!

**Comment by yangwao** [comment]
pay 80 usd
merge then when it will builds 👀

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 4.48 USD/DOT ~ 17.857 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xc85aa03ade2438aa5751e0bff3066c27a796e77c38ee7cfa0c8d858419ea1da4](https://polkadot.subscan.io/extrinsic/0xc85aa03ade2438aa5751e0bff3066c27a796e77c38ee7cfa0c8d858419ea1da4)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by prury** [comment]
buy action is being show as a sale: 

https://github.com/kodadot/nft-gallery/assets/36627808/7aeb16f7-cce2-4db8-b377-5f02cd0e634f

**Comment by exezbcz** [comment]
@prury yes, I think its because we don't have buy event in collection - its the same from the collection perspective. Lets make this followup and I will design a tag for that

**Comment by prury** [comment]
mobile spacing: 
image

**Comment by hassnian** [comment]
@prury the margin is still there and is the same as the desktop version
CleanShot 2023-09-01 at 08 46 23@2x

it looks like less because the columns on mobile are not shown
CleanShot 2023-09-01 at 08 48 34@2x

@exezbcz what should be the margin then ?

**Comment by exezbcz** [comment]
In case of the desktop, would it be possible to put it at the same row as the filters please?
image
- On mobile, its okay to leave them like this, 24px top bottom margin would be better tho
image


- Please make the filters jump to the second row if they don't fit 
image

**Pull request merged**